lib/OpenQA/Utils.pm

```diff
@@ -1184,18 +1185,28 @@ sub rand_range { $_[0] + rand($_[1] - $_[0]) }
 sub in_range   { $_[0] >= $_[1] && $_[0] <= $_[2] ? 1 : 0 }
 
 sub set_listen_address {
-    my ($port) = @_;
-
-    if ($ENV{MOJO_LISTEN}) {
-        return;
-    }
+    my $port = shift;
 
+    return if $ENV{MOJO_LISTEN};
     my @listen_addresses = ("http://127.0.0.1:$port");
-    if (IO::Socket::IP->new(Listen => 5, LocalAddr => '::1')) {
-        push(@listen_addresses, "http://[::1]:$port");
-    }
 
-    $ENV{MOJO_LISTEN} = join(',', @listen_addresses);
+    # Check for IPv6
+    push @listen_addresses, "http://[::1]:$port" if IO::Socket::IP->new(Listen => 5, LocalAddr => '::1');
+
+    $ENV{MOJO_LISTEN} = join ',', @listen_addresses;
+}
+
+sub service_port {
+    my $service = shift;
+
+    my $base = $ENV{OPENQA_BASE_PORT} ||= 9526;
+
+    if    ($service eq 'webui')       { return $base }
+    elsif ($service eq 'websocket')   { return $base + 1 }
+    elsif ($service eq 'livehandler') { return $base + 2 }
+    elsif ($service eq 'scheduler')   { return $base + 3 }
```

Comment:
A hash storing those offsets would be nicer.

Reply:
I don't really see much value in making the logic more complex.

Reply:
It doesn't look cleaner.
```perl
    my $offsets = {webui => 0, websocket => 1, livehandler => 2, scheduler => 3};
    croak unless exists $offsets->{$service};
    return $base + $offsets->{$service};
```

Reply:
Let's vote on it, thumbs up on the snippet above means change it, thumbs down keep the `if/elsif` version. 😄

Reply:
I prefer the if/else version, it's more explicit even if it's more code.

Reply:
I can live with both. I tend to use hashes instead of large if/elsifs, but IIMHO 4 is still ok.
In case of the hash, I think I'd write:
```
my $offsets = {
    webui => 0,
    websocket => 1,
    livehandler => 2,
    scheduler => 3,
};
```

Hashes are also ideal if this information could be used elsewhere.

Reply:
Ok, you win... this time. 😉